plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/util/Util.java

```diff
@@ -65,24 +60,4 @@ public static String getFolder( String path ) {
     }
     return path.substring( 0, path.lastIndexOf( "/" ) );
   }
-
-  public static void rawElementMassage( Object rawElement ) {
-    //Performs conversion of raw elements from drag and drop to something the system understands
-    if ( rawElement instanceof VFSFile ) {
-      VFSFile vfsFile = (VFSFile) rawElement;
-      if ( vfsFile.getPath().startsWith( "hcp://" ) ) {
-        //HCP Files come back like this "hcp://hcp.hitachi.com:443/path1/filename" and for some reason this does not
-        // work with HCP.  We work around this by converting the URL to pvfs like this
-        // ( ie: pvfs://myconnectionName/path1/filename ).  The displayed name is what we want.
-        try {
-          URI uri = new URI( vfsFile.getPath() );
-          vfsFile.setPath( ConnectionFileProvider.SCHEME + "://" + vfsFile.getConnection() + uri.getPath() );
```

Comment:
Completely not needed anymore we are not translating from Provider URI <--> PVFS URI ie s3:// <--> pvfs://, just using pvfs://

Reply:
short answer: yes the default s3 logic is still in place and not effected.

detailed answer:
We aren't blocking anyone from manually entering a Provider or Native URI (s3://... or gs://..) URI into ktrs/kjb. It will be passed straight to KettleVFS.java. The issue is will the Pentahos VFS Providers have all the necessary file system options for authentication. 

in the case for S3 we have the legacy "default" logic still in place here:
https://github.com/pentaho/big-data-plugin/blob/10347702d81ad9786390c8578957673cf5849693/s3-vfs/src/main/java/org/pentaho/s3common/S3CommonFileSystem.java#L78-L97

In on of the S3 Connection Details, a user has to check the related box.

AWS S3 still provides the various many options for credentials passing such as environment variables, system props,... https://docs.aws.amazon.com/cli/v1/userguide/cli-chap-authentication.html which the Pentaho S3 VFS Provider code honors.

Reply:
On second look, I see what you are talking about. The "feature" bringing up a Native S3 URI in PDI File Open Save Browser was working in 9.3.0.4 only because of the Penatho "default" logic for Connection Details. Since 9.4, that feature was broken during to the feature of the PDI File Open Browser conversions from JavaScript to Java. Currently, no VFS URI (s3, gs, pvfs, etc...) entered launched from A PDI step will show up and navigate to the folder properly.

However, that "feature" is a "one off" only possible due to  
Pentaho Default "default" logic for Connection Details in S3. This wouldn't work for Google cloud storage (gs) or any VFS that needs file system options to pull data.

A work around would just use the pvfs connection that is flagged as default to get an URI, then just replace the "pvfs://<connectionName>" with "s3://". 

To accomplish this 9.3.0.4 functionality, We would have to tweak the tree traversal logic to not error out if it can't update the left lane folder structure since the path in the navigation bar does not belong to a registered metastore connection details. And still allow the top operation button such as up directory and what not. 

I can bring this to up Ramaiz.

Reply:
I assume S3 was a special case because it didn't need any additional configuration. The driver can use the on-disk credentials of the running user. Similar for SMB on windows. I'm not aware of any other vfs types that don't need any configuration.

Reply:
It "shouldn't" be a big effort to allow native S3 and SMB to display in the, but that either will get done in https://hv-eng.atlassian.net/browse/PDI-20094 or I follow up story.